records/2023-01-26.bounce-rate-automated-emergency-brake.md

```diff
@@ -130,72 +134,82 @@ having
+    \text{length} ( \text{sliding notifications} )
+}$$
 
 
 This above bounce rate is not an exact number but it an approximation of what the bounce rate is likely to be within a sliding window.
 
-#### Keys Task
+## Keys Task
 
 If we wanted to see all the keys for ALL the services - we should make an AWS ECS task that we can kick off.\
 Instead as an optimization, we should only seed the keys for a service where:
 1. A user from the service has logged into the ADMIN
 1. We get a message from the API for a certain service
 
 In both the above scenarios we will kick off a celery task that will see the bounce rate data. We need to ensure that if users of the service login, or we send multiple jobs through the API - we are not constantly starting the celery task to seed the bounce rate.
-In order for us to keep track, we will need to add a key to services table indicating if we have started to seed the bounce rate for that service. This will prevent duplication of jobs.
 
-#### Redis Memory Consideration
+## Redis Memory Consideration
+
+We are currently using 500mb on our Redis cluster which is assigned 1G of RAM. An increase in the number of keys means we might need to upgrade the cluster. [Current memory usage estimations and analysis](https://github.com/cds-snc/notification-attic/pull/31) and recent [Redis memory usage](https://gcdigital.slack.com/archives/CV38DBNVA/p1685372981086749) shows that, for the time being, this is not something we need to consider right now, but as Notify grows we will need to keep this in mind.
+
+# Additional considerations
+
+## Staggered roll-out
+To prevent suspending services that may not actually be causing issues with the overall bounce rate, it is suggested to roll out this feature in stages.
 
-We are currently using 500mb on our Redis cluster which is assigned 1G of RAM. An increase in the number of keys means we might need to upgrade the cluster.
+### Version 1
+The first stage would include tracking bounces, calculating bounce rates, and displaying them on service dashboards. It would also include providing bounce rate guidance for services.
 
-## Additional considerations
+### Version 1.5
+The second stages would include bug fixes, and any adjustments to the way we calculate bounce rate.
 
-### Staggered roll-out
-To prevent suspending services that may not actually be causing issues with the overall bounce rate, it is suggested to roll out this feature in stages.  The first stage would include tracking bounces, calculating bounce rates, and displaying them on service dashboards. It would also include providing bounce rate guidance for services. Once the calculations have been confirmed to work satisfactorily with live data, the second stage can be rolled out which will include warning and suspending services.
+### Version 2
+Once the calculations have been confirmed to work satisfactorily with live data, the third stage can be rolled out which will include warning and suspending services.
```

Comment:
Let's remove the "first stage" "second stage" "third stage" terms and just refer to the releases by version number.

```suggestion
Once the calculations have been confirmed to work satisfactorily with live data, we can roll out warning and suspending services that exceed a bounce rate of 10%. 
```